src/execution/operator/aggregate/physical_window.cpp

```diff
@@ -333,6 +334,29 @@ static idx_t FindNextStart(const BitArray<W> &mask, idx_t l, idx_t r) {
 	return r;
 }
 
+template <typename W>
+static idx_t FindPrevStart(const BitArray<W> &mask, const idx_t l, idx_t r) {
+	while (l < r) {
+		//	If r-1 is aligned with the start of a block, and the block is blank, then skip backwards one block.
+		const auto block = mask.GetBlock(mask.Block(r - 1));
+		auto shift = mask.Shift(r - 1);
+		if (!block && !shift) {
+			r -= mask.BITS_PER_WORD;
```

Comment:
This code is not run, could you add a test case that triggers this?

Reply:
Good catch - the code as written was not correct. I wrote a multi-block test that has an initial empty block.